persist/fs/files.go

```diff
@@ -232,6 +245,30 @@ func FilesetBefore(filePathPrefix string, namespace ts.ID, shard uint32, t time.
 	return filesBefore(matched, t)
 }
 
+// DeleteInactiveFilesets deletes any filesets that are not currently owned by the namespace
+func DeleteInactiveFilesets(filePathPrefix string, namespace ts.ID, activeShards []uint32) error {
+	var toDelete []string
+	activeDirNames := make(map[string]struct{})
+	namespaceDirPath := NamespaceDirPath(filePathPrefix, namespace)
+	allShardDirs, err := findSubDirectoriesAndPaths(namespaceDirPath)
+	if err != nil {
+		return nil
+	}
+
+	//create shard set, might also be useful to just send in as strings?
```

Comment:
nit: could you drop the question in the comment. Minor nit: convention is to put a space after `//`, ie `// Create ...` instead of `//create ...`

Reply:
+1